src/Autocomplete/src/AutocompleteResultsExecutor.php

```diff
@@ -51,14 +56,56 @@ public function fetchResults(EntityAutocompleterInterface $autocompleter, string
 
         $nbPages = (int) ceil($paginator->count() / $queryBuilder->getMaxResults());
 
-        $results = [];
-        foreach ($paginator as $entity) {
-            $results[] = [
-                'value' => $autocompleter->getValue($entity),
-                'text' => $autocompleter->getLabel($entity),
-            ];
+        $options = [];
+        $optgroups = [];
+
+        if (null !== $groupBy = $autocompleter->getGroupBy()) {
+            if (\is_string($groupBy)) {
+                $groupBy = new PropertyPath($groupBy);
+            }
+
+            if ($groupBy instanceof PropertyPathInterface) {
+                $accessor = $this->propertyAccessor;
+                $groupBy = function ($choice) use ($accessor, $groupBy) {
+                    try {
+                        return $accessor->getValue($choice, $groupBy);
+                    } catch (UnexpectedTypeException) {
+                        return null;
+                    }
+                };
+            }
+        }
+
+        if (\is_callable($groupBy)) {
```

Comment:
If it's not a callable at this point, that would be an error, right? I'm imagining that we check for `null` and `string` above (and for string, you're converting into a callable). So we could say:

```php
if (!\is_callable($groupBy)) {
    // then throw an exception
}
```